web3.js/src/message/index.ts

```diff
@@ -15,18 +19,27 @@ export type MessageHeader = {
   numReadonlyUnsignedAccounts: number;
 };
 
+/**
+ * An address table lookup used to load additional accounts
+ */
+export type MessageAddressTableLookup = {
+  accountKey: PublicKey;
+  writableIndexes: Array<number>;
+  readonlyIndexes: Array<number>;
+};
+
 /**
  * An instruction to execute by a program
  *
  * @property {number} programIdIndex
- * @property {number[]} accounts
- * @property {string} data
+ * @property {number[]} accountKeyIndexes
+ * @property {Uint8Array} data
  */
-export type CompiledInstruction = {
+export type MessageCompiledInstruction = {
   /** Index into the transaction keys array indicating the program account that executes this instruction */
   programIdIndex: number;
   /** Ordered indices into the transaction keys array indicating which accounts to pass to the program */
-  accounts: number[];
-  /** The program input data encoded as base 58 */
-  data: string;
+  accountKeyIndexes: number[];
```

Comment:
`MessageCompiledInstruction` is meant to be the successor to `CompiledInstruction`. Encoding the data is unnecessary and inefficient so that was the main reason I wanted to move away from `CompiledInstruction`. The `accounts` -> `accountKeyIndexes` isn't as big of a deal but I feel that it's more clear and descriptive.